akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowGraphCompileSpec.scala

```diff
@@ -227,10 +226,10 @@ class FlowGraphCompileSpec extends AkkaSpec {
 
     "build with implicits and variance" in {
       FlowGraph.closed() { implicit b ⇒
-        def appleSource = b.add(Source(PublisherProbe[Apple]))
-        def fruitSource = b.add(Source(PublisherProbe[Fruit]))
-        val outA = b add Sink(SubscriberProbe[Fruit]())
-        val outB = b add Sink(SubscriberProbe[Fruit]())
+        def appleSource = b.add(Source(TestPublisher.probe[Apple]))
```

Comment:
hmm, thinking a bit, what if the probe just extends Source[T, Unit] instead? Then it can be used in the DSL without the extra hassle, it does not need a materialization trip to get to the probe. I think it is not hard to implement (and the same for the Subscriber side)